src/json-serializer/json-serializer.spec.ts

```diff
@@ -47,12 +47,125 @@ describe("serializer Tests", () => {
     }
   );
 
-  it("Serializer should throw error",
+  it('Serializer should throw error',
     () => {
-      const array = ["abc"];
+      const array = ['abc'];
       expect(function () {
         Cdp4JsonSerializer.serialize(array);
-      }).toThrow(new Error("The type string not supported yet"));
+      }).toThrow(new Error('The type string not supported yet'));
+    }
+  );
+
+  it('ValueArray with special characters is serialized and deserialized without errors',
+    () => {
+      const jsonString = `{"widget": {
+                "debug": "on",
+                "window": {
+                    "title": "Sample Konfabulator Widget",
+                    "name": "main_window",
+                    "width": 500,
+                    "height": 500
+                },
+                "image": { 
+                    "src": "Images/Sun.png",
+                    "name": "sun1",
+                    "hOffset": 250,
+                    "vOffset": 250,
+                    "alignment": "center"
+                },
+                "text": {
+                    "data": "Click Here",
+                    "size": 36,
+                    "style": "bold",
+                    "name": "text1",
+                    "hOffset": 250,
+                    "vOffset": 100,
+                    "alignment": "center",
+                    "onMouseUp": "sun1.opacity = (sun1.opacity / 100) * 90;"
+                }
+            }}`;
+      const xmlString = `<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
In this XML string please remove all copyrighted material and real author names.

Reply:
Then we already have this issue in Java and C# SDK that needs to be fixed. Will you mind this fix without PR once we agree here on the fix?

Reply:
Agreed, feel free to push changes directly to dev on those repos, here changes look good,